### PR TITLE
Added Rabbit Base Service file and refactoring 

### PR DESCRIPTION
**Pull request by JoooJooo**

### CODE REVIEW

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/NodeExecutor.java [hunk 1]
```diff
@@ -74,7 +74,6 @@ public NodeResponse call() throws Exception {
             Document executionDetailsDoc = executionDocumentService.fetchExecutionDetails(this.node.getObjectId(),this.node.getExecutionId());
             try {
                 String executionDataJson= ((Document)executionDetailsDoc.get("executionData")).toJson();
```
Comment (devvsda) [inline shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/NodeExecutor.java]: Why this again ?

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/application/ShepherdApplication.java [hunk 2]
```diff
@@ -1,15 +1,15 @@
-package com.devsda.platform.shephardcore.application;
```
Comment (devvsda) [inline shephard-core/src/main/java/com/devsda/platform/shepherdcore/application/ShepherdApplication.java]: +1 for correcting the name. Please do the same for configuration files also.

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+public class RabbitMqOperations {
+
+    private static final Logger log = LoggerFactory.getLogger(RabbitMqOperations.class);
+    private static Connection connection;
+    private static Channel channel;
+
+    public static void createQueueConnection(String userName, String password,
+                                            String virtualHost, String hostName, Integer portNumber)
+            throws IOException, TimeoutException {
+        if(connection==null) {
```
Comment (devvsda) [inline shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java]: Add proper loggings.

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            ConnectionFactory factory = new ConnectionFactory();
+            factory.setUri(connectionString);
+            connection = factory.newConnection();
+        }
+        if(channel==null){
+            channel = connection.createChannel();
+        }
+
+        return connection;
```
Comment (devvsda) [inline shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java]: Use intellij indetation to correct all touched files.

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+                MessageProperties.PERSISTENT_TEXT_PLAIN,
+                messageBodyBytes);
+    }
+
+    public GetResponse pullMessage(String queueName, Boolean autoAck) throws IOException {
+        GetResponse response = channel.basicGet(queueName, autoAck);
+        if (response == null) {
+            log.info(String.format("No message received from queue %s",queueName));
+            return null;
+        } else {
```
Comment (devvsda) [inline shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java]: No need of this else. Directly write return response.

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+            return;
+        }
+
+        byte[] messageBodyBytes = message.getBytes();
+        channel.basicPublish(exchangeName, routingKey,
+                MessageProperties.PERSISTENT_TEXT_PLAIN,
+                messageBodyBytes);
+    }
+
+    public GetResponse pullMessage(String queueName, Boolean autoAck) throws IOException {
```
Comment (devvsda) [inline shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java]: Where is visibility timeout and number of messages configuration ?

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+            log.info(String.format("No message received from queue %s",queueName));
+            return null;
+        } else {
+            return response;
+        }
+    }
+
+
+    public void ackReceivedMessage(long deliveryTag, Boolean multipleAckEnabled) throws IOException {
+        System.out.println(" [*] Waiting for messages. To exit press CTRL+C");
```
Comment (devvsda) [inline shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java]: Remove System.out.println(). Use logger

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+        channel.basicAck(deliveryTag, multipleAckEnabled);
+    }
+
+    public static String fetchMessageFromGetResponse(GetResponse response){
```
Comment (devvsda) [inline shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java]: Use <T> T to convert into desired POJO. Make it generic.

File: shepherd-model/src/main/java/com/devsda/platform/shepherd/model/ExecuteWorkflowRequest.java [hunk 9]
```diff
@@ -34,6 +34,19 @@
     @JsonProperty("initialPayload")
     private String initialPayload;
 
+    public ExecutionData getExecutionData() {
```
Comment (devvsda) [inline shepherd-model/src/main/java/com/devsda/platform/shepherd/model/ExecuteWorkflowRequest.java]: Why this ?

File: shepherd-model/src/main/java/com/devsda/platform/shepherd/model/ExecutionData.java [hunk 10]
```diff
@@ -0,0 +1,33 @@
+package com.devsda.platform.shepherd.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Map;
+
+
+public class ExecutionData {
```
Comment (devvsda) [inline shepherd-model/src/main/java/com/devsda/platform/shepherd/model/ExecutionData.java]: Need to remove this.

File: shepherd-model/src/main/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQOperations.java [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+import com.rabbitmq.client.*;
+
+import javax.print.URIException;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.util.concurrent.TimeoutException;
+
+public class RabbitMQOperations {
```
Comment (devvsda) [inline shepherd-model/src/main/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQOperations.java]: Why two files of same code ?

File: shepherd-model/src/main/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQOperations.java [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+        GetResponse response = channel.basicGet("first_queue", false);
+        if (response == null) {
+            // No message retrieved.
+        } else {
+            AMQP.BasicProperties props = response.getProps();
+            byte[] body = response.getBody();
+            String responseMessage = new String(body);
+            System.out.println(responseMessage);
+            long deliveryTag = response.getEnvelope().getDeliveryTag();
+            System.out.println("delivery Tag : "+ deliveryTag);
```
Comment (devvsda) [inline shepherd-model/src/main/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQOperations.java]: Remove sysout. Use logs.

File: shepherd-model/src/test/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQTest.java [hunk 13]
```diff
@@ -0,0 +1,106 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.util.concurrent.TimeoutException;
+
+public class RabbitMQTest {
```
Comment (devvsda) [inline shepherd-model/src/test/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQTest.java]: Change class name.

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/NodeExecutor.java (hunk 1 above)
Comment (JoooJooo) [reply shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/NodeExecutor.java]: executionData field is in Mongo Document.

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java (hunk 3 above)
Comment (JoooJooo) [reply shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java]: sure.

Comment (JoooJooo) [reply shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java]: done.

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java (hunk 6 above)
Comment (JoooJooo) [reply shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java]: checking how to apply that

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java (hunk 5 above)
Comment (JoooJooo) [reply shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java]: done.

File: shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java (hunk 7 above)
Comment (JoooJooo) [reply shephard-core/src/main/java/com/devsda/platform/shepherdcore/service/queueservice/RabbitMqOperations.java]: removed

File: shepherd-model/src/main/java/com/devsda/platform/shepherd/model/ExecuteWorkflowRequest.java (hunk 9 above)
Comment (JoooJooo) [reply shepherd-model/src/main/java/com/devsda/platform/shepherd/model/ExecuteWorkflowRequest.java]: removed

File: shepherd-model/src/main/java/com/devsda/platform/shepherd/model/ExecutionData.java (hunk 10 above)
Comment (JoooJooo) [reply shepherd-model/src/main/java/com/devsda/platform/shepherd/model/ExecutionData.java]: removed

File: shepherd-model/src/main/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQOperations.java (hunk 11 above)
Comment (JoooJooo) [reply shepherd-model/src/main/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQOperations.java]: On Models I was testing it , Will remove this.

File: shepherd-model/src/main/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQOperations.java (hunk 12 above)
Comment (JoooJooo) [reply shepherd-model/src/main/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQOperations.java]: This file will not get checked in so no need

File: shepherd-model/src/test/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQTest.java (hunk 13 above)
Comment (JoooJooo) [reply shepherd-model/src/test/java/com/devsda/platform/shepherd/rabbitmq/RabbitMQTest.java]: Sure,.

**Pull request merged**